stdlib/REPL/src/REPL.jl

```diff
@@ -776,11 +776,46 @@ find_hist_file() = get(ENV, "JULIA_HISTORY",
 
 backend(r::AbstractREPL) = r.backendref
 
+# Allows an external package to add hooks into the code loading.
+# The hook should take a Vector{Symbol} of package names and
+# return true if all packages could be installed, false if not
+# to e.g. install packages on demand
+const install_packages_hooks = Any[]
+
 function eval_with_backend(ast, backend::REPLBackendRef)
+    if !isempty(install_packages_hooks)
+        check_for_missing_packages_and_run_hooks(ast)
```

Comment:
This should probably run in the backend. It could even use the existing `repl_ast_transforms` hook (not a "transform" per se, but no big deal).

Reply:
Ok. Moved to the backend, but given that the `modules_to_be_loaded` parser resides here, it seemed like a misuse to place that on `repl_ast_transforms` given that if IJulia/Pluto etc. wanted to do a different install hook they'd probably want to go after the modules have been identified by `modules_to_be_loaded` so keeping `install_packages_hooks` would be necessary anyway. These use models are a bit imaginary though.. my thinking might be off